lib/beaker-puppet/install_utils/puppet_utils.rb

```diff
@@ -86,6 +90,64 @@ def get_puppet_collection(agent_version = 'latest')
           collection
         end
 
+        # Determine the puppet collection that matches a given version of the puppet-agent
```

Comment:
I don't think it makes sense to combine them anyway -- versions that come from `puppet --version` and `facter aio_agent_version` are always going to be different in puppet4, because they're not describing the same packages. Once PC1 is totally retired, the verisions for puppet and puppet-agent will keep pace with one another into the future, though.

Reply:
Makes sense. Thanks.

Reply:
We have a policy for this going forward in 6.Y, but several 5.Y packages got their Zs out of sync between puppet and agent, as well